packages/codeeditor/src/viewer.ts

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import { StackedLayout, Widget } from '@lumino/widgets';
+import { CodeEditor } from './editor';
+import { CodeEditorWrapper } from './widget';
+
+export class CodeViewerWidget extends Widget {
```

Comment:
Since this will be part of the public API, a question about the exported name.

Is this functionality specific to code? Or should we indicate this can be used for any text content? If so maybe the "Code" part in `CodeViewerWidget` could be misleading.

Reply:
This was actually a point of discission when I first created this widget in Elyra. I had originally called it `TextViewerWidget` but didn't quite like it and after some discussion in a Elyra dev call we settled on `CodeViewerWidget`.

Our reasoning was that this wraps the CodeEditor and specifically enabled code highlighting based on mime type, so even though it can be used for any text, its purpose is for displaying code. I'm not too stuck on the name, but I definitely prefer `CodeViewerWidget` more than `TextViewerWidget`